life.js

```diff
@@ -0,0 +1,291 @@
+    ctx.strokeStyle = BORDER_COLOUR;
+    ctx.stroke();
+    ctx.strokeRect(0, 0, canvasHeight, canvasWidth);
+}
+
+function draw() {
+    ctx.clearRect(0, 0, canvasWidth, canvasHeight);
+    drawCells();
+    drawGrid();
+    if (displayNeighbours)
+        drawNeighbourCount();
+}
+
+//TODO: Fix centering/colours/size of displayed numbers
+function drawNeighbourCount() {
+    ctx.font = "20px Arial";
+    ctx.fillStyle = NEIGHBOUR_COUNT_COLOUR;
+
+    for (let c = 0; c < CELL_COLUMN_COUNT; c++) {
+        for (let r = 0; r < CELL_ROW_COUNT; r++) {
+            ctx.fillText(cells[c][r].neighbours, c * CELL_WIDTH, r * CELL_HEIGHT + CELL_HEIGHT);
+        }
+    }
+}
+
+/* -------------------------------- game functions ------------------------------- */
+function calculateNeighbourCount() {
+    for (let c = 0; c < CELL_COLUMN_COUNT; c++) {
+        for (let r = 0; r < CELL_ROW_COUNT; r++) {
+            let neighbourCount = 0;
+            //check surrounding and count alive cells
+            for (let XOffset = -1; XOffset < 2; XOffset++) {
+                for (let YOffset = -1; YOffset < 2; YOffset++) {
+                    //check only 8 surrounding cells
+                    if (!(XOffset == 0 && YOffset == 0)) {
+                        //make sure checked cell is within grid bounds
+                        if (c + XOffset >= 0 && c + XOffset < CELL_COLUMN_COUNT && r + YOffset >= 0 && r + YOffset < CELL_ROW_COUNT) {
+                            if (cells[c + XOffset][r + YOffset].alive == true)
+                                neighbourCount++
+                        }
+                    }
+                }
+            }
+            cells[c][r].neighbours = neighbourCount;
+        }
+    }
+}
+
+function nextGeneration() {
+    for (let c = 0; c < CELL_COLUMN_COUNT; c++) {
+        for (let r = 0; r < CELL_ROW_COUNT; r++) {
+            //determine if cell will live to next generation
+            if (cells[c][r].neighbours < 2)
+                cells[c][r].aliveNext = false;
+            else if (cells[c][r].neighbours == 2 && cells[c][r].alive == true)
+                cells[c][r].aliveNext = true;
+            else if (cells[c][r].neighbours == 3)
+                cells[c][r].aliveNext = true;
+            else if (cells[c][r].neighbours > 3)
+                cells[c][r].aliveNext = false;
+        }
+    }
+    for (let c = 0; c < CELL_COLUMN_COUNT; c++) {
+        for (let r = 0; r < CELL_ROW_COUNT; r++) {
+            cells[c][r].alive = cells[c][r].aliveNext
+            cells[c][r].aliveNext = false;
+        }
+    }
+    calculateNeighbourCount();
+}
+
+//Controls game logic and timing on next generation
+function gameLoop() {
+    if (playing || iterateGeneration) {
+        nextGeneration();
+        draw();
+        console.log("next generation");
+        iterateGeneration = false;
+    }
+}
+
+//Initialises the game by drawing the play area
+function startGame() {
+    draw();
+}
+
+startGame();
+

```

Comment:
A bit redundant here, can just call `draw();`
```suggestion
draw();
```